tensorflow/compiler/tf2tensorrt/convert/ops/unary_ops.cc

```diff
@@ -0,0 +1,72 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#if GOOGLE_CUDA && GOOGLE_TENSORRT
+
+#include "tensorflow/compiler/tf2tensorrt/convert/convert_nodes.h"
+#include "tensorflow/compiler/tf2tensorrt/convert/op_converter_registry.h"
+#include "tensorflow/compiler/tf2tensorrt/convert/ops/layer_utils.h"
+
+namespace tensorflow {
+namespace tensorrt {
+namespace convert {
+
+class ConvertUnary : public OpConverterBase<ConvertUnary> {
+ public:
+  explicit ConvertUnary(OpConverterParams* params)
+      : OpConverterBase<ConvertUnary>(params) {}
+
+  static constexpr std::array<DataType, 3> AllowedDataTypes() {
+    return {DataType::DT_FLOAT, DataType::DT_HALF};
+  }
+
+  static constexpr std::array<InputArgSpec, 1> InputSpec() {
+    return std::array<InputArgSpec, 1>{
+        InputArgSpec::Create("my_unary", TrtInputArg::kBoth)};
+  }
+
+  Status Validate() {
+    const auto& op = params_->node_def.op();
+    if (UnaryOperationMap()->find(op) == UnaryOperationMap()->end()) {
+      return errors::Unimplemented("Unary op: ", op, " not supported");
+    }
+
+    return CheckInputsWeights(*params_, {{"x", false}});
```

Comment:
This is checked in the base class via the InputSpec() you defiend above. So you shouldn't need this function call and also not need to declare `CheckInputsWeightrs` in the header

Reply:
Good point! Fixed.